spec/requests/users_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Users API', type: :request do
+  let(:valid_user_params) do
+    {
+      user: {
+        name: Faker::Name.name,
+        email: Faker::Internet.email,
+        password: 'Password123+',
+        password_confirmation: 'Password123+'
+      }
+    }
+  end
+
+  describe 'Signup' do
+    let(:headers) do
+      {
+        'ACCEPT' => 'application/json'
+      }
+    end
+
+    context 'valid user params' do
+      it 'creates a user account for valid params' do
+        expect { post '/api/v1/signup.json', params: valid_user_params, headers: }.to change(User, :count).from(0).to(1)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:created)
+        expect(JSON.parse(response.body)['errors']).to be_empty
+      end
+    end
+
+    context 'invalid user params' do
+      # Common expectations bundled for a DRY code.
+      def expectations(invalid_user_params)
+        expect { post '/api/v1/signup.json', params: invalid_user_params, headers: }.not_to change(User, :count)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors']).not_to be_empty
+      end
```

Comment:
While I like this idea, I think it's a bit overkill for now. Repeating expects is pretty standard in tests and only when it comes to 4 or 5 (or greater) tests should we move it to a separate method

Reply:
Alright, I'll embrace duplication for now and I'll leave this for a refactoring.

Reply:
Done